modulus/models/afno/modembed.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Type
+
+import torch
+from torch import Tensor, nn
+
+
+class PositionalEmbedding(nn.Module):
+    """
+    A module for generating positional embeddings based on timesteps.
+
+    Parameters:
+    -----------
+    num_channels : int
+        Number of channels for the embedding.
+    """
+
+    def __init__(self, num_channels: int):
+        super().__init__()
+        self.num_channels = num_channels
+
+        freqs = torch.pi * torch.arange(
+            start=1, end=self.num_channels // 2 + 1, dtype=torch.float32
+        )
+        self.register_buffer("freqs", freqs)
+
+    def forward(self, x: Tensor):
```

Comment:
-> Tensor

Reply:
Fixed in https://github.com/NVIDIA/modulus/pull/547/commits/f634dfdcaae1cc73e0e67568843c7fd7fd5d1d09.